### PR TITLE
Text2Cypher custom prompt: doc, example and bug fix

### PR DESCRIPTION
**Pull request by stellasia**
# Description

- Add `Text2CypherTemplate` to the API doc
- Add an example to demonstrate how to use custom prompt for Text2Cypher retrievers
- Fix tests

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [x] Documentation update
- [ ] Project configuration change

## Complexity

Complexity: Low

## How Has This Been Tested?
- [x] Unit tests
- [ ] E2E tests
- [x] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] Unit tests have been updated
- [ ] E2E tests have been updated
- [x] Examples have been updated
- [ ] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed
- [x] CHANGELOG.md updated if appropriate

### CODE REVIEW

File: tests/unit/retrievers/test_text2cypher.py [hunk 1]
```diff
@@ -310,3 +311,32 @@ def test_t2c_retriever_with_custom_prompt_bad_prompt_params(
     llm.invoke.assert_called_once_with(
         """This is a custom prompt. test ['example A', 'example B']"""
     )
+
+
+@patch("neo4j_graphrag.retrievers.base.Retriever._verify_version")
+@patch("neo4j_graphrag.retrievers.text2cypher.get_schema")
```
Comment (willtai) [inline tests/unit/retrievers/test_text2cypher.py:317]: Does the order matter here? Why are the orders for the other tests changed?

Comment (stellasia) [reply tests/unit/retrievers/test_text2cypher.py:317]: Yes, the order matters, when using a decorator, the mocks are introduced in a given order: see [here](https://docs.python.org/3/library/unittest.mock.html#quick-guide) (you need to scroll down a bit to `def test(MockClass1, MockClass2):` and read note below)

Comment (stellasia) [reply tests/unit/retrievers/test_text2cypher.py:317]: Though I need to check to make sure my update is using the proper one (I'm always getting confused about which one comes first)

Comment (stellasia) [reply tests/unit/retrievers/test_text2cypher.py:317]: I checked again, and I think it's alright now.

**Pull request merged**